functional_tests/scylla_operator/test_functional.py

```diff
@@ -462,6 +462,30 @@ def test_scylla_operator_pods(db_cluster: ScyllaPodCluster):
         [pods_info['name'] for pods_info in scylla_operator_pods if pods_info['status'] != 'Running'])
     assert not not_running_pods, f'There are pods in state other than running: {not_running_pods}'
 
+    # Cover https://github.com/scylladb/scylla-operator/issues/408:
+    # Operator shouldn't run as StatefulSet, should be Deployment
+    for pods_info in scylla_operator_pods:
+        pod_replicaset_names = get_pod_replicaset_names(db_cluster=db_cluster,
+                                                        namespace=SCYLLA_OPERATOR_NAMESPACE,
+                                                        pod_name=pods_info['name'])
+        assert len(pod_replicaset_names) > 0, "Failed to get replicaSet names"
+
+        for replicaset_name in pod_replicaset_names:
+            rs_owner_references = get_replicaset_name_and_kind(db_cluster=db_cluster,
+                                                               namespace=SCYLLA_OPERATOR_NAMESPACE,
+                                                               replicaset_name=replicaset_name)
+            assert len(rs_owner_references) > 0, "Failed to get replicaSet names"
+
+            for rs_owner_reference in rs_owner_references:
+                assert rs_owner_reference["name"], "Failed to get replicaSet owner references name"
+                assert rs_owner_reference["kind"] == 'Deployment', \
+                    f"Replicaset kind should be 'Deployment', but it's '{rs_owner_reference['kind']}'"
+
+                deployment = db_cluster.k8s_cluster.kubectl(f"get deployment {rs_owner_reference['name']} -o yaml",
```

Comment:
why this cross check is needed ? i.e. you got it by kind, why do we need to get the configuration of it ? (especially if we don't even read anything out of it)

Reply:
It's like full cycle - check that we can fetch deployment data. Suggested by @vponomaryov

Reply:
It may or may not be. It is optional. If we remove it we won't lose anything, because we do not test K8S itself here. So, we should assume that parent resources always exists.

Reply:
removed